### DISCUSSION TITLE
How to run Python code in Elsa 3.1

### DISCUSSION THREAD

**Discussion by PPAdminUser** [elsa-workflows/elsa-core#5209] [Q&A] [answered]
Hello
I have been testing Elsa to see if it can cover the requirements of my company.
And one of the requirement we have is to run Python scripts.
According to the documentation Elsa should support it but i can't find where in the Studio i can implement it.

When building workflows under the Scripting tag all i can see is the C# and Javascript nodes.
does anyone know what i am doing wrong?

I am running Elsa in Docker using the command on the Documentation page..

Thank you

**Comment by glime-ai** [comment, marked as answer]
Hello,

Elsa Workflows does support scripting, including C# and JavaScript expressions, as you've noticed in the Scripting section. However, as of my last update, Python scripting was not yet fully tested and may not be available directly within the Elsa Studio visual designer. This could be the reason why you are not seeing Python nodes alongside C# and JavaScript.

If Python scripting is a critical requirement for your company, you might want to check the official documentation for any updates on Python support or look into the possibility of extending Elsa with custom activities that can execute Python scripts. Additionally, you could reach out to the Elsa community or the maintainers for guidance on how to proceed with Python scripting.

Since you are running Elsa in Docker, make sure you are using the latest version of the Elsa Docker image and that your environment is correctly set up to support Python. If Python support has been recently added or updated, it may require specific environment configurations or additional dependencies.

For the most accurate and up-to-date information, I recommend checking the official Elsa documentation, GitHub repository issues, or forums for any announcements or discussions related to Python scripting support.

If you need further assistance, please provide more details about your Docker setup and the version of Elsa you are using, and I can give you more specific guidance.

**Comment by mohdali** [comment]
@PPAdminUser for docker, please refer to [`ElsaServerAndStudio.DockerFile`](https://github.com/elsa-workflows/elsa-core/blob/main/docker/ElsaServerAndStudio.Dockerfile) for an example of how to configure python.

```dockerfile
# Install Python 3.11
RUN apt-get update && apt-get install -y --no-install-recommends \
    python3.11 \
    python3.11-dev \
    libpython3.11 \
    python3-pip && \
    rm -rf /var/lib/apt/lists/*

# Set PYTHONNET_PYDLL environment variable
ENV PYTHONNET_PYDLL /usr/lib/aarch64-linux-gnu/libpython3.11.so
```
